happi/cli.py

```diff
@@ -396,23 +396,48 @@ def edit(ctx, name: str, edits: list[str]):
 
 
 @happi_cli.command()
-@click.argument('item_names', nargs=-1)
+@click.option('--glob/--regex', 'use_glob', default=True,
+              help='use glob style (default) or regex style search terms. '
+              r'Regex requires backslashes to be escaped (eg. at\\d.\\d)')
+@click.argument('search_criteria', nargs=-1)
 @click.pass_context
-def load(ctx, item_names: list[str]):
+def load(
+    ctx: click.Context,
+    use_glob: bool,
+    search_criteria: list[str]
+):
+    """ Checks if the search string contains a glob pattern. If so,
+    it's sent to the search_parser() function. """
+
+    item_names = []
+
+    if '*' not in (' '.join(search_criteria)):
+        item_names = search_criteria
+    else:
+        final_results = search_parser(
+            client=get_happi_client_from_config(ctx.obj),
+            use_glob=use_glob,
+            search_criteria=search_criteria,
+        )
```

Comment:
I think this actually doesn't accomplish what we had agreed upon in the dev discussion.   What we were looking for was the logical OR of the search items, aka the union of the individual search terms.  What this does is take all the search terms and gives the logical AND of the search terms.  

What we discussed would be implemented by gathering the results of each search term individually, then loading the union of the results.  In pseudocode it'd be something like:
```python
results = set()
for search_term in search_criteria:
    res = search_parser(
        client=<the client>,
        use_glob=use_glob, search_criteria=search_term
    )
    results.add(res)

item_names = [r['name'] for r in results]
```

Reply:
Sorry, didn't see this before I responded to the previous comment. Seems like a `load_parser()` function should be created.